### PR TITLE
Add `InverseShift` extension

### PR DESCRIPTION
**Pull request by jinjiren**
This PR aims to add support for InverseShift extension, which is compatible to the `inv` learning rate policy in Caffe. 

As mentioned in #4546, having support for various learning rate schedule policies will make reproducing results of existing papers easier.

I'm planning to add more `Shift` extensions, however, before that I would like to see whether this one is proper for merge.

### CODE REVIEW

**Comment by bkvogel** [comment]
Thank you for the PR! Please fix flake8 errors.

**Comment by jinjiren** [comment]
@bkvogel I fixed the flake8 errors (checked locally). Sorry that I forgot to use the `flake8` installed by `hacking` package, instead I had used the common `flake8` (which gives no errors to the previous code...).

**Comment by bkvogel** [comment]
I think it looks good. My only comment is that there are not any tests. Although it is not required, if you can think of any ways to test the expected behavior, please consider adding tests. For an example, see the existing extensions tests such as https://github.com/chainer/chainer/blob/master/tests/chainer_tests/training_tests/extensions_tests/test_exponential_shift.py

**Comment by kmaehashi** [comment]
👍Please add it to the docs also.
https://github.com/chainer/chainer/blob/master/docs/source/reference/training.rst#optimizer-behavior-control

**Comment by bkvogel** [comment]
@jinjiren could you also please fix the conflict in `__init__.py`? thanks!

**Comment by jinjiren** [comment]
Added unittests and docs.

**Comment by bkvogel** [comment]
Jenkins, test this please.

**Comment by bkvogel** [comment]
Jenkins, test this please.

**Comment by bkvogel** [comment]
LGTM. The tests passed except one that looks unrelated to your PR.
